### PR TITLE
Fix/1731

### PR DESCRIPTION
**Pull request by manasnisar**
## Changes
Getting config from MPT
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/blobber.go [hunk 1]
```diff
@@ -388,10 +388,17 @@ func (sc *StorageSmartContract) commitBlobberRead(t *transaction.Transaction,
 			"can't get related allocation: %v", err)
 	}
 
+	conf, err = getConfig(balances)
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/blobber.go]: you already have the config,[ line 340 above](https://github.com/0chain/0chain/blob/58c79ee907524b37ea79303e3e0086d59155c9a1/code/go/0chain.net/smartcontract/storagesc/blobber.go#L340). you don't need it again.

File: code/go/0chain.net/smartcontract/storagesc/challengepool.go [hunk 2]
```diff
@@ -189,7 +189,14 @@ func (ssc *StorageSmartContract) createChallengePool(t *transaction.Transaction,
 	// create related challenge_pool expires with the allocation + challenge
 	// completion time
 	var cp *challengePool
-	cp, err = ssc.newChallengePool(alloc.ID, t.CreationDate, alloc.Until(),
+	conf, err := getConfig(balances)
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/challengepool.go]: The config is avaliable further up the stack in [newAllocationRequestInternal](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/allocation.go#L218). Pass it down in parameters, don't get it from the MPT again.

File: code/go/0chain.net/smartcontract/storagesc/allocation.go [hunk 3]
```diff
@@ -1542,8 +1542,15 @@ func (sc *StorageSmartContract) finalizeAllocation(
 			"allocation already finalized")
 	}
 
+	conf, err := getConfig(balances)
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/storagesc/allocation.go:1471]: We call the config already in [finishAllocation](https://github.com/0chain/0chain/blob/58c79ee907524b37ea79303e3e0086d59155c9a1/code/go/0chain.net/smartcontract/storagesc/allocation.go#L1735). We want to replace this call in finishAllocation with the one you just created here in `finalizeAllocation`. So remove getConfig from finsihAllocation and add as a parameter to finishAllocation.

**Review by Sriep — DISMISSED** [review]

File: code/go/0chain.net/smartcontract/storagesc/models.go [hunk 4]
```diff
@@ -1171,8 +1171,7 @@ func (sa *StorageAllocation) validateEachBlobber(
 }
 
 // Until returns allocation expiration.
-func (sa *StorageAllocation) Until() common.Timestamp {
-	var conf = Config{}
+func (sa *StorageAllocation) Until(conf *Config) common.Timestamp {
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/models.go]: Pass the MaxChallengeCompletionTime as param should be enough as this method doesn't care about the other configs.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/models.go (hunk 4 above)
Comment (manasnisar) [reply code/go/0chain.net/smartcontract/storagesc/models.go]: Made the changes

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/1731  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/1731  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/1731  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/1731  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/1731  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/1731  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | batch-update  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/1731  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/1731  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/1731  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**